sklearn/neural_network/tests/test_mlp.py

```diff
@@ -876,3 +876,20 @@ def test_mlp_loading_from_joblib_partial_fit(tmp_path):
     # finetuned model learned the new target
     predicted_value = load_estimator.predict(fine_tune_features)
     assert_allclose(predicted_value, fine_tune_target, rtol=1e-4)
+
+
+@pytest.mark.parametrize("Estimator", [MLPClassifier, MLPRegressor])
+def test_preserve_feature_names(recwarn, Estimator):
+    # Non-regression test for #24846
+
+    pd = pytest.importorskip("pandas")
+
+    X = pd.DataFrame(
+        data=[[i, i] for i in range(10)], columns=["colname_a", "colname_b"]
+    )
+    y = pd.DataFrame(data=[[1] for i in range(10)], columns=["colname_y"])
```

Comment:
Can this be a pandas series since we will only be using `y["colname_y"]`? Concretely:

```python
y = pd.Series([1] * 10, name="colname_y")
```